packages/graphql-typescript-definitions/src/filesystem/graphql-filesystem.ts

```diff
@@ -62,7 +62,7 @@ export abstract class AbstractGraphQLFilesystem extends EventEmitter {
     config: GraphQLProjectConfig,
   ): boolean;
 
-  emit(event: 'change:schema', path: string): boolean;
+  emit(event: 'change:schema', input: string | string[]): boolean;
```

Comment:
Some fs watchers collate multiple changes. The built-in watcher's lib (Chokidar) doesn't appear to support this, but it wouldn't be super difficult to implement.

However, this is more of a "future API improvement" change; the reworked `generateDocumentTypes` short-circuits any duplicate work caused by multiple emits.